futures-io/src/lib.rs

```diff
@@ -427,6 +450,31 @@ mod if_std {
         deref_async_write!();
     }
 
+    impl<T: ?Sized + Unpin> AsyncWrite for Arc<T>
+    where
+        for<'a> &'a T: AsyncWrite,
+    {
+        fn poll_write(self: Pin<&mut Self>, cx: &mut Context<'_>, buf: &[u8])
+            -> Poll<Result<usize>>
+        {
+            Pin::new(&mut &**self).poll_write(cx, buf)
+        }
+
+        fn poll_write_vectored(self: Pin<&mut Self>, cx: &mut Context<'_>, bufs: &[IoSlice<'_>])
+            -> Poll<Result<usize>>
+        {
+            Pin::new(&mut &**self).poll_write_vectored(cx, bufs)
+        }
+
+        fn poll_flush(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<()>> {
+            Pin::new(&mut &**self).poll_flush(cx)
+        }
+
+        fn poll_close(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<()>> {
+            Pin::new(&mut &**self).poll_close(cx)
+        }
```

Comment:
```suggestion
        deref_async_write!();
```

Reply:
This doesn't compile - note there is a slight difference between what this does and what `deref_async_write!()` does:

* `Pin::new(&mut &**self).poll_write(cx, buf)`
* `Pin::new(&mut **self).poll_write(cx, buf)`

Reply:
Oops, you are right. Sorry for the wrong review!